diff --git a/lwe/snark/r1cs_lattice_snark.hpp b/lwe/snark/r1cs_lattice_snark.hpp
--- a/lwe/snark/r1cs_lattice_snark.hpp
+++ b/lwe/snark/r1cs_lattice_snark.hpp
@@ -96,15 +96,15 @@ void gen_q_mat(const r1cs_constraint_system<libff::Fr<ppT>> &cs,
             H_qs[i] = std::move(qap_inst.Ht);
             Z_s[i] = qap_inst.Zt;
-            A_prefix[i].reserve(num_inputs + 1);
+            A_prefix[i].resize(num_inputs + 1);
             std::copy_n(std::begin(A_qs[i]), num_inputs + 1,
                         std::begin(A_prefix[i]));
-            B_prefix[i].reserve(num_inputs + 1);
+            B_prefix[i].resize(num_inputs + 1);
             std::copy_n(std::begin(B_qs[i]), num_inputs + 1,
                         std::begin(B_prefix[i]));
-            C_prefix[i].reserve(num_inputs + 1);
+            C_prefix[i].resize(num_inputs + 1);
             std::copy_n(std::begin(C_qs[i]), num_inputs + 1,
                         std::begin(C_prefix[i]));
         }
